sdk/provisioning/Generator/src/Specifications/PostgreSqlSpecification.cs

```diff
@@ -29,5 +29,6 @@ protected override void Customize()
         AddNameRequirements<PostgreSqlDatabaseResource>(min: 1, max: 63, lower: true, upper: true, digits: true, hyphen: true);
         AddNameRequirements<PostgreSqlFirewallRuleResource>(min: 1, max: 128, lower: true, upper: true, digits: true, hyphen: true, underscore: true);
         AddNameRequirements<PostgreSqlVirtualNetworkRuleResource>(min: 1, max: 128, lower: true, upper: true, digits: true, hyphen: true);
+        AddNameRequirements<PostgreSqlFlexibleServerResource>(min: 3, max: 63, lower: true, digits: true, hyphen: true);
```

Comment:
Do we need the other "flexible server" resources as well?

* PostgreSqlFlexibleServerFirewallRule
* PostgreSqlFlexibleServerDatabase
* PostgreSqlFlexibleServerActiveDirectoryAdministrator

?

Reply:
I'm happy to add any naming rules we find.  I don't see any clear requirements for [firewall rules](https://learn.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/flexibleservers/firewallrules?pivots=deployment-language-bicep#flexibleserversfirewallrules), [databases](https://learn.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/flexibleservers/databases?pivots=deployment-language-bicep#flexibleserversdatabases), or [admins](https://learn.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/flexibleservers/administrators?pivots=deployment-language-bicep#flexibleserversadministrators) though.

Reply:
The main reason I ask is because we have them for the non-flexible server directly above this line.

Reply:
Yes - those are defined at https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftdbforpostgresql.  I don't see any equivalents for flexible server.